### PR TITLE
Implements editing of sample names

### LINKED ISSUE #69: Paired file picker - support for parsing/editing group names
Issue pulled out of #55. UI will be expected to parse filenames user selects and guess the group(sample) names for each pair of files. Users would then need to be able to edit them (for when the guess is wrong). This data would need to be passed to the workflow requiring changes to the workflow to support these inputs. 

One possibility for modeling this data would be to create a custom CWL data type that represents named pairs of files. Example:
```
"named_read_pairs": [
  {
    "name":"SRR099967",
    "read1": {
        "class": "File",
        "path": "...SRR099967_1.fastq.gz"
      },
     "read2": {
        "class": "File",
        "path": "...SRR099967_2.fastq.gz"
      }
   }
  ],
```
A fallback would be to pass the filenames as an array if a data structure like the above is intractable in CWL.

### PR DESCRIPTION
**Pull request by dleehr**
- Replaces static text label with an editable text field for sample names
- Refactors data structure for the file pairs in the sample, introducing a new class in the utility class: `FASTQSample`
- Prevents auto-generation of sample name when user has provided one

Fixes #69

### CODE REVIEW

**Comment by johnbradley** [comment]
The UI for editing sample pair names works well for the most part.
However if you remove a sample (ie remove two files that are in a pair) it loses any custom pair names below that in the list.
For example a set of pairs like so:

- Name: SAMPLE1
  - ❌ FILEA_R1.fastq.gz
  - ❌ FILEA_R1.fastq.gz
- Name: SAMPLE2
  - ❌ FILEB_R1.fastq.gz
  - ❌ FILEB_R1.fastq.gz

If you delete FILEA_R1.fastq.gz and FILEA_R1.fastq.gz the second sample reverts to FILEB:
- Name: FILEB
  - ❌ FILEB_R1.fastq.gz
  - ❌ FILEB_R1.fastq.gz

This is pretty nit-picky but thought I would mention it.

**Comment by johnbradley** [comment]
Are we creating a separate issue to handle changes to the sample name parser handle the issue mentioned here: https://github.com/Duke-GCB/bespin-ui/issues/69#issuecomment-354452114 ?

**Comment by dleehr** [comment]
>  if you remove a sample (ie remove two files that are in a pair) it loses any custom pair names below that in the list.

Good observation. Instead of the current behavior (deleting one file causes everything below it to shift up and re-pair), do you suppose it's better to leave the gap, where the next chosen file will go?

**Comment by dleehr** [comment]
> Are we creating a separate issue to handle changes to the sample name parser handle the issue mentioned here: #69 (comment) ?

Yeah I think we'll need to - I don't think there's enough information there to determine what the sample name is.

**Comment by johnbradley** [comment]
> Instead of the current behavior (deleting one file causes everything below it to shift up and re-pair), do you suppose it's better to leave the gap, where the next chosen file will go?

If we went that route we would need to convey to the user which cell we plan on populating. 
Imagine the user removes one file from two different samples, then adds another file.
Which pair would it add to?

**Comment by dleehr** [comment]
> Which pair would it add to?

The current behavior is that it finds the first empty slot and adds it there. We could highlight that sample somehow to show where the next file would go, but I think this is quickly becoming its own issue :)

File: app/utils/fastq-file-item-list.js [hunk 1]
```diff
@@ -21,54 +20,201 @@ function extractSampleName(name, separators) {
...
+    return this.get('fileItems.length') === this.get('size');
+  }),
+  isEmpty: Ember.computed('fileItems.length', function() {
+    return this.get('fileItems.length') === 0;
+  }),
+  ddsFiles: Ember.computed.mapBy('fileItems', 'ddsFile'),
+  fileItemsLength: Ember.computed.alias('fileItems.length'),
+  inputFiles: Ember.computed.mapBy('fileItems', 'inputFile'),
+  includesFileItem(fileItem) {
+    return this.indexOfFileItem(fileItem) != -1;
```
Comment (johnbradley) [inline app/utils/fastq-file-item-list.js]: Perhaps `!==` to be consistent with other comparisons here?

File: tests/integration/components/questionnaire/fastq-file-pair-row-test.js [hunk 2]
```diff
@@ -19,24 +21,4 @@ test('it renders both files from the pair', function(assert) {
...
-    pair.set('file1', null);
-    pair.set('file2', {name: 'pairFile2'});
-    this.render(hbs`{{questionnaire/fastq-file-pair-row pair}}`);
-    assert.equal(this.$('.file-group-file').length, 1);
-
-    // Do not render the first file when not set
-    assert.equal(this.$('.file-group-file').eq(0).text().trim(), 'pairFile2');
-  });
-});
+// TODO: test that editing names are reflected
```
Comment (johnbradley) [inline tests/integration/components/questionnaire/fastq-file-pair-row-test.js]: Stray TODO comment.

**Pull request merged**